### DISCUSSION TITLE
use tootctl on another .env file

### DISCUSSION THREAD

**Discussion by fedi-will** [mastodon/mastodon#22990] [Q&A]
My situation:
 - installed two instances on one server
 - Debian 11, no Docker, installed from scratch according to official docs

I have a running instance on a .social domain, all works fine.
Now I installed another instance on a .de domain on the same server. (according to [this ](https://nfgworld.com/multiple-mastodon-instances-on-a-single-server/)article)

My problem:
I cannot use tootctl to manage the new .de instance
If I do: 
`RAILS_ENV=production bin/tootctl accounts create admin --email admin@example.com --confirmed --role Owner`
it complains about user/mail already exists because it uses the .social .env.production file (I guess)

How can I use the .de instance's .env.production.de file???

**Comment by StrangeMinds1960** [comment]
Shouldn't you be using 'RAILS_ENV=production.de'?

**Reply by fedi-will** [reply]
@StrangeMinds1960 thanks for looking into my problem
Your suggestion results in this error:
```
config.eager_load is set to nil. Please update your config/environments/*.rb files accordingly:
  * development - set it to false
  * test - set it to false (unless you use a tool that preloads your test environment)
  * production - set it to true
```
along with a longer stack trace

**Comment by fedi-will** [comment]
Spent last 2 hours trying to create new config file _config/environments/production.de.rb_ and get my .de instance to use this config. 
Failed in many ways. 

I might need to use an env var RAILS_ENV_FILE=production.de to tell rails what .env file to use...
Maybe too rails specific, asking around on Stackoverflow

Looks like not being too easy to run 2 instances on same server...

**Reply by StrangeMinds1960** [reply]
I followed the same instructions you did a while back to create two instances on one server, never worked, but I can't remember why

**Reply by StrangeMinds1960** [reply]
Another thing worth noting, those instructions are 4.5 years old now, much has changed since then I suspect

**Reply by fedi-will** [reply]
> those instructions are 4.5 years old now, much has changed since then I suspect

That's also what I think. It took me a bit of tweaking to get the second instance running. 
But now I am completely stuck with the 2 problems I have asked here. 
Maybe it's not worth the effort. Still great to have one instance up and running
